docs/source/data/data_catalog.md

```diff
@@ -30,6 +30,40 @@ shuttles:
   load_args:
     engine: openpyxl # Use modern Excel engine (the default since Kedro 0.18.0)
 ```
+
+### Configuring dataset parameters in `catalog.yml`
+
+The dataset configuration in `catalog.yml` is defined as follows:
+1. The Top-level key is the dataset name used as a dataset identifier in the catalog - `shuttles`, `weather` in the example below.
+2. The next level includes multiple keys. The first one is the mandatory key, `type,` which defines the type of dataset to use.
+The rest of the keys are dataset properties and vary depending on the implementation.
+To get the extensive list of dataset properties, refer to {py:mod}`The kedro-datasets package documentation <kedro-datasets:kedro_datasets>` and navigate to the `__init__` method of the target dataset.
+3. Some dataset properties can be further configured depending on the libraries underlying the dataset implementation.
```

Comment:
What's an example of this and how this different from 2.? Where can user find information about this?

Reply:
The example is in the following line. I've extended it a bit for clarity. The difference is that some of the parameters require referring to the underlying library methods to get the full set of the parameters accepted. It is not clear for some users, so we wanted to explicitly show that in the docs.

https://github.com/kedro-org/kedro/issues/3919#issuecomment-2189050360